mail_embed_image/models/ir_mail_server.py

```diff
@@ -83,48 +52,41 @@ def build_email(
             body_alternative=body_alternative,
             subtype_alternative=subtype_alternative,
         )
-        return self._build_email_replace_img_src(result)
+        if fileparts:
+            for fpart in fileparts:
+                result.attach(fpart)
+        return result
 
-    def _build_email_replace_img_src(self, email):
-        """ Given a message, find it's img tags and if they
-        are URLs, replace them with cids.
-        """
-        for part in email.walk():
-            if part.get_content_type() == 'text/html':
-                body = part.get_payload(decode=True)
-                if not body or body == '\n':
-                    continue
-                root = self._build_email_process_img_body(
-                    fromstring(body), email)
-                # encodestring will put a newline every 74 char
-                part.set_payload(encodestring(tostring(root)))
-        return email
+    def _build_email_replace_img_src(self, html_body):
+        """Replace img src with base64 encoded image."""
+        if not html_body:
+            return html_body
 
-    def _build_email_process_img_body(self, root, email):
-        base_url = self.env['ir.config_parameter'].get_param(
-            'web.base.url')
-        for img in root.xpath(
-                ".//img[starts-with(@src, '%s/web/image')]"
-                "| .//img[starts-with(@src, '/web/image')]" % (base_url)):
-            image_path = img.get('src').replace(base_url, '')
-            with self._fetch_image(image_path) as (endpoint, arguments):
-                # now go ahead and call the endpoint and fetch the data
-                response = endpoint.method(**arguments)
-                if not response or response.status_code != 200:
-                    logger.warning('Could not get %s', img.get('src'))
-                    continue
-                cid = uuid.uuid4().hex
-                filename_rfc2047 = encode_header_param(cid)
-                filepart = MIMEImage(response.data)
-                # TODO check if filepart exists (do not attach twice)
-                filepart.set_param('name', filename_rfc2047)
-                filepart.add_header(
-                    'Content-Disposition',
-                    'inline',
-                    cid=cid,
-                    filename=filename_rfc2047,
-                )
-                # attach the image into the email as attachment
-                email.attach(filepart)
-                img.set('src', 'cid:%s' % (str(cid)))
-        return root
+        root = fromstring(html_body)
+        images = root.xpath("//img")
+        fileparts = []
+        for img in images:
+            src = img.get("src")
+            if src and not src.startswith("data:") and not src.startswith("base64:"):
+                try:
+                    response = requests.get(src, timeout=10)
```

Comment:
the reason the original did the fetching in a somewhat roundabout way is that what you do here allows malicious users to craft emails to themselves and have Odoo fetch arbitrary resources from the internal network. Can be harmless, can be catastrophic depending on what's accessible from there

Reply:
May be harmless...
Yes, probably.
Is functionally really expected, yes too.
Thanks @imlopes for this code.

Reply:
thanks for the feedback @hbrunn @fliot 
Im really not very comfortable with this kind of module. I've tried to keep as much as I could the original code but lot of changes between the original version and the new one on odoo side.

Reply:
well as it is I really don't think it should be published.

But what was the problem? Currently you'd call https://github.com/OCA/OCB/blob/16.0/odoo/addons/base/models/ir_http.py#L95 which relieves you from constructing the routing map yourself, and you only need to create a dummy request bound to the public user in a pretty similar way the v10 version does it